src/BlockTypes/ProductGalleryThumbnails.php

```diff
@@ -67,22 +108,31 @@ protected function render( $attributes, $content, $block ) {
 				if ( $product_gallery_images && $post_thumbnail_id ) {
 					$html                 = '';
 					$number_of_thumbnails = isset( $block->context['thumbnailsNumberOfThumbnails'] ) ? $block->context['thumbnailsNumberOfThumbnails'] : 3;
+					$mode                 = $block->context['mode'] ?? '';
 					$thumbnails_count     = 1;
 
 					foreach ( $product_gallery_images as $product_gallery_image_html ) {
-						if ( $thumbnails_count > $number_of_thumbnails ) {
+						// Limit the number of thumbnails only in the standard mode (and not in dialog).
+						if ( 'standard' === $mode && $thumbnails_count > $number_of_thumbnails ) {
 							break;
 						}
 
-						$processor = new \WP_HTML_Tag_Processor( $product_gallery_image_html );
+						// If not in dialog and it's the last thumbnail and the number of product gallery images is greater than the number of thumbnails settings output the View All markup.
+						if ( 'standard' === $mode && $thumbnails_count === $number_of_thumbnails && count( $product_gallery_images ) > $number_of_thumbnails ) {
```

Comment:
This conditional is quite hard to read. I would suggest extracting them into their own little functions.

Reply:
Thank you, you are right. I have moved both conditionals to separate functions.